src/main/java/Replay.java

```diff
@@ -112,7 +112,7 @@ public void tick() {
                         Manipulator.nanoOffsetsToLocalDateTime(d.data,localStartTime,skipNanoOffset,speed);
                         Manipulator.generateID(d.data);
                         // TODO: can add modifier call here for realtime modifications.  Not sure how performant these would be.
-                        Main.apiHandler.postData(d);
+                        Main.apiHandler.postData(d,lastSentIdx);
```

Comment:
the index being passed would be the last index attempted to post rather than the current one. 
index also doesnt seem to be used.